default.nix

```diff
@@ -0,0 +1,86 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake , boost , itk , swig, libGL
+, buildPython ? true, python ? null, numpy ? null
+, buildLibs ? true,
+}:
+
+let
+  ver = "6.2.0";
+  # Uncomment and update this if you want to build from GitHub
+  #sha256 = "0fzfm3ki0v2g09jqjakrq3asz69snmnydm3iwcqi6k9rabgn4g7z";
+
+in stdenv.mkDerivation {
+  name = "stir-${ver}";
+
+  src = lib.cleanSource ./.;
+  # Uncomment and update this if you want to build from GitHub
+#   src = fetchFromGitHub {
+#     owner = "UCL";
+#     repo = "STIR";
+#     rev = "rel_${ver}";
+#     inherit sha256;
+#   };
+
+  buildInputs = [ boost cmake itk /*openmpi*/ ];
+  propagatedBuildInputs = [ swig ]
+    ++ lib.optional buildPython [ python numpy ];
+
+  # This is a hackaround because STIR requires source available at runtime.
+  setSourceRoot = ''
+    actualSourceRoot=;
+    for i in *;
+    do
+        if [ -d "$i" ]; then
+            case $dirsBefore in
+                *\ $i\ *)
+
+                ;;
+                *)
+                    if [ -n "$actualSourceRoot" ]; then
+                        echo "unpacker produced multiple directories";
+                        exit 1;
+                    fi;
+                    actualSourceRoot="$i"
+                ;;
+            esac;
+        fi;
+    done;
+
+    # "Install" location for source
+    sourceRoot=$prefix/src
+    mkdir -p $sourceRoot
+    # Put the actual source there
+    cp -r $actualSourceRoot -T $sourceRoot
+  '';
+  cmakeFlags = [
+    "-DBUILD_TESTING=ON"
+    "-DGRAPHICS=PGM"
+    "-DSTIR_MPI=OFF"
+    "-DSTIR_OPENMP=${if stdenv.isDarwin then "OFF" else "ON"}"
+   ] ++ lib.optionals (buildLibs) [
+    "-DBUILD_SHARED_LIBS=ON"
```

Comment:
I doubt this works.

Reply:
Why? I'll have to check. I believe I needed it to build SIRF against STIR, but made it optional here in the hopes it builds faster and with a smaller closure.

Reply:
There's #6 (our almost oldest issue!) . It's possible that it works on Linux, but definitely not on other systems.

You surely don't need it for SIRF (VM, docker, GHA and all builds that I know of don't use it).

It should be a lot faster to build though, so if it works, you could keep it (but change the name of the variable!)